src/mlpack/methods/lmnn/lmnn_function_impl.hpp

```diff
@@ -395,17 +468,44 @@ void LMNNFunction<MetricType>::Gradient(const arma::mat& transformation,
   // Apply metric over dataset.
   transformedDataset = transformation * dataset;
 
-  // Calculate norm of change in transformation.
   double transformationDiff = 0;
-  if (!transformationOld.is_empty())
+  if (!transformationOld.is_empty() && iteration++ % range == 0)
```

Comment:
`transformationDiff` is used every iteration, so shouldn't we avoid adding the `&& iteration++ % range == 0` here?

Reply:
Done!